integration-tests/main.go

```diff
@@ -34,6 +35,7 @@ func GenerateConfig(name string, jsonData []byte) error {
 	if writeErr := ioutil.WriteFile(name, out.Bytes(), 0777); writeErr != nil {
 		fmt.Printf("Error writing temp config file: %v", writeErr)
 	}
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Why we need a timeout here ???

Reply:
I was having issues while testing but forget to remove it before the PR. Good find!